### PR TITLE
307513: Add managed ID to postgres AD group on pre-migration stage

### PR DESCRIPTION
**Pull request by jeevankuduvaravindran**
# **What this PR does / why we need it**:
This PR contains the below changes to add Team MI to postgres writer AD group
- Added script to add MI to AD group
- Updated Grant scripts to accommodate the changes
- Updated flux manifests in pre-deploy stage for ffc-demo-claim-service
[AB#307513](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/307513)

# **Special notes for your reviewer**
This PR has a series of changes. 
https://github.com/DEFRA/adp-flux-core/pull/40
https://github.com/DEFRA/adp-infrastructure-core/pull/137/files
https://github.com/DEFRA/adp-pipeline-common/pull/132/files

Updates to flux manifest automation to generate the pre-deploy manifests for all services will be done in a follow-up PR

# Testing
*Any relevant testing information and pipeline runs.*

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

**Comment by DaRoszko** [comment]
Great PR. My question is that how much of what is added to the Flux Manifests needs to be added or is missing from the scaffolder for Flux Manifests? @kbabigumira  @aadev1  @jeevankuduvaravindran

**Review by kbabigumira — APPROVED** [review]

File: services/ffc/ffc-demo/ffc-demo-claim-service/pre-deploy-kustomize.yaml [hunk 1]
```diff
@@ -30,6 +30,9 @@ spec:
       POSTGRES_DB: "ffc-demo-claim"
       POSTGRES_PORT: "5432"
       POSTGRES_SCHEMA_NAME: "public"
+      #These variables will need to be populated after onboarding automation when flux manifests are generated
```
Comment (DaRoszko) [inline services/ffc/ffc-demo/ffc-demo-claim-service/pre-deploy-kustomize.yaml:33]: Great PR. My question is that how much of what is added to the Flux Manifests needs to be added or is missing from the scaffolder for Flux Manifests? @kbabigumira @aadev1 @jeevankuduvaravindran

**Review by DaRoszko — APPROVED** [review]

**Review by DaRoszko — CHANGES_REQUESTED** [review]
See comment previously

File: services/ffc/ffc-demo/ffc-demo-claim-service/pre-deploy-kustomize.yaml (hunk 1 above)
Comment (jeevankuduvaravindran) [reply services/ffc/ffc-demo/ffc-demo-claim-service/pre-deploy-kustomize.yaml:33]: @DaRoszko For now, the below variables needs to come from scaffolder which are missing now.
 postgres AD groups, postgres DB name and schema name.

**Review by DaRoszko — APPROVED** [review]

**Pull request merged**